clients/da-vinci-client/src/main/java/com/linkedin/davinci/helix/AbstractPartitionStateModel.java

```diff
@@ -263,6 +263,11 @@ protected void removePartitionFromStoreGracefully() {
     metaSystemStoreReplicaStatusNotifier.ifPresent(
         systemStoreReplicaStatusNotifier -> systemStoreReplicaStatusNotifier
             .drop(storeConfig.getStoreVersionName(), partition));
+    if (!ingestionBackend.getStorageService()
+        .getStorageEngineRepository()
+        .hasLocalStorageEngine(storeConfig.getStoreVersionName())) {
+      ingestionBackend.closeStoreIngestionTask(storeConfig);
```

Comment:
I guess this is the important part of the fix, right?

I wonder if this ordering is ideal? Should consumption be halted prior to the storage getting dropped, rather than after?

I also wonder if it is the right pattern to have all of these sequential interactions with the `ingestionBackend`?

Is there a use case for calling just some but not all of these functions? Regardless of whether such a use case exists, for this use case here, should we consider having the `ingestionBackend` expose a single function for doing all of the cleanups related to a partition getting unsubbed (and potentially other side-effects if that was the last partition)?

Reply:
The consumption will be halted in above line: 
    ingestionBackend.dropStoragePartitionGracefully(
        storeConfig,
        partition,
        getStoreConfig().getPartitionGracefulDropDelaySeconds());
in the above.

Also, there is removeCustomizedState() here in between so I am not sure if we can put all these logics into one method in the ingestionBackend, probably we can move it to the end as I see no intersection between the logic inside that method and other logics.

 probably we can have the ingestionBackend.dropPartitionGracefully(topic, partition, whetherToDropEmptyStorageEngine) include the logic to close SIT.
The 3rd arg is to accommodate the state for II where it still maintains the metadata partition for future ingestion. 
Regardless of 3rd arg, I think we can drop SIT when all data partitions are closed.
What do you think?

Reply:
I mean, you know this part of the code best so I'm asking questions but don't know what the "right" answers are 😅

I do want us to think about II in the context of this code. Is each interaction with the `ingestionBackend` going to trigger an IPC to the child process? If yes, then it may be easier to reason about if all related operations and necessary side-effects were encapsulated in a single function and single IPC. Otherwise doing multiple ones may make the system harder to reason about (i.e. in terms of the number of permutations of parent process state and child process state).

Reply:
Yeah, I guess so, what you said makes me to think more about II and I found that current status won't let II close SIT when last data partition is dropped actually. I will try to change that part and send out the change.

Reply:
I merged the operation into IngestionBackend implementation. IsolatedIngestionBackend is is just a delegator so for II it will do one remote call and invoke the same method in DefaultIngestionBackend.